jubatus/core/anomaly/lof_storage.cpp

```diff
@@ -271,7 +271,25 @@ bool lof_storage::put_diff(const lof_table_t& mixed_diff) {
       lof_table_[it->first] = it->second;
     }
   }
+
+  // Create a set of removed (unlearned) rows since get_diff.
+  unordered_set<std::string> removed_ids;
+  for (lof_table_t::const_iterator it = lof_table_diff_.begin();
+      it != lof_table_diff_.end(); ++it) {
+    if (is_removed(it->second)) {
+      removed_ids.insert(it->first);
+    }
```

Comment:
I think removed ID (``kdist < 0``) will be propagated eternally.
I think should check existence in ``mixed_diff`` or remove ID from ``lof_table_diff`` in L.262.